### PR TITLE
5.0: Update `django.contrib.admin.widgets`

### PR DESCRIPTION
**Pull request by JaeHyuckSa**
# I have made things!
Update stubs for `django.contrib.admin.widgets` for Django 5.0.

- [x]  `django.contrib.admin.widgets`
  - [x]  `django.contrib.admin.widgets.BaseAdminDateWidget` was added
  - [x]  `django.contrib.admin.widgets.AdminDateWidget` was changed
  - [x]  `django.contrib.admin.widgets.BaseAdminTimeWidget` was added
  - [x]  `django.contrib.admin.widgets.AdminTimeWidget` was changed
  - [x]  `django.contrib.admin.widgets.RelatedFieldWidgetWrapper.choices` was added

## Related issues

Refs
- https://github.com/typeddjango/django-stubs/issues/1493

Upstream PR
`django.contrib.admin.widgets.RelatedFieldWidgetWrapper.choices`
- https://github.com/django/django/pull/16943

`django.contrib.admin.widgets.BaseAdminDateWidget`
`django.contrib.admin.widgets.AdminDateWidget` 
`django.contrib.admin.widgets.BaseAdminTimeWidget` 
`django.contrib.admin.widgets.AdminTimeWidget`
- https://github.com/django/django/pull/16961

### CODE REVIEW

File: django-stubs/contrib/admin/widgets.pyi [hunk 1]
```diff
@@ -21,12 +21,24 @@ class FilteredSelectMultiple(forms.SelectMultiple):
         choices: _Choices = ...,
     ) -> None: ...
 
-class AdminDateWidget(forms.DateInput):
+class BaseAdminDateWidget(forms.DateInput):
+    class Media:
```
Comment (sobolevn) [inline django-stubs/contrib/admin/widgets.pyi]: CC @flaeppe 

I think that defining such classes may cause an error when defining subclasses of widgets.

It is close to out `Meta` case.

Comment (saJaeHyukc) [reply django-stubs/contrib/admin/widgets.pyi:25]: @sobolevn I have some sympathy for what you said. In the case of this part meta, I will delete it and commit it :)

Comment (sobolevn) [reply django-stubs/contrib/admin/widgets.pyi]: Let's ignore it for now, we can later decide what to do.

Comment (saJaeHyukc) [reply django-stubs/contrib/admin/widgets.pyi]: I revised what you said :)

**Review by sobolevn — APPROVED** [review]

**Pull request merged**